api-proxy/src/server.mjs

```diff
@@ -0,0 +1,101 @@
+import * as http from 'http'
+import createLogger from './logging.mjs'
+import {handleReq} from './reqhandler.mjs'
+
+const nl = '\n'
+
+const parseUrl = (protocol, req) => {
+  const {host} = req.headers
+  if (!host) {
+    throw Err('no-host', 'missing Host header', {status: 400})
+  }
+  if (host.startsWith(protocol+'://')) {
```

Comment:
Yes, since a client can stuff anything they want into any header, but anything invalid should cause this to throw, which would then show up as an uncaught error.